### PR TITLE
Document System.Buffers.ReadOnlySequence

### PR DESCRIPTION
**Pull request by carlossanlop**
Source here: https://source.dot.net/#System.Memory/System/Buffers/ReadOnlySequence.cs

### CODE REVIEW

File: xml/System.Buffers/ReadOnlySequenceSegment`1.xml [hunk 1]
```diff
@@ -47,7 +47,7 @@
       </AssemblyInfo>
       <Parameters />
       <Docs>
-        <summary>To be added.</summary>
+        <summary>Initializes a new read-only sequence segment instance.</summary>
```
Comment (mairaw) [inline xml/System.Buffers/ReadOnlySequenceSegment`1.xml]: Style guide: https://github.com/dotnet/dotnet-api-docs/wiki/Summary%3A-Constructor

```suggestion
        <summary>Initializes a new instance of the <see cref="T:System.Buffers.ReadOnlySequenceSegment`1" /> struct.</summary>
```

File: xml/System.Buffers/ReadOnlySequenceSegment`1.xml [hunk 2]
```diff
@@ -23,7 +23,7 @@
   </Base>
   <Interfaces />
   <Docs>
-    <typeparam name="T">To be added.</typeparam>
+    <typeparam name="T">The type of the elements in the sequence segment.</typeparam>
```
Comment (mairaw) [inline xml/System.Buffers/ReadOnlySequenceSegment`1.xml]:
```suggestion
    <typeparam name="T">The type of the elements in the read-only sequence segment.</typeparam>
```

File: xml/System.Buffers/ReadOnlySequence`1.xml [hunk 3]
```diff
@@ -28,7 +28,7 @@
     </Attribute>
   </Attributes>
   <Docs>
-    <typeparam name="T">To be added.</typeparam>
+    <typeparam name="T">The type of the elements in the sequence.</typeparam>
```
Comment (mairaw) [inline xml/System.Buffers/ReadOnlySequence`1.xml]:
```suggestion
    <typeparam name="T">The type of the elements in the read-only sequence.</typeparam>
```

File: xml/System.Buffers/ReadOnlySequence`1.xml [hunk 4]
```diff
@@ -148,12 +148,22 @@ The consumer is expected to manage the lifetime of memory until <see cref="T:Sys
...
+        <exception cref="T:System.ArgumentNullException"><paramref name="startSegment" /> or <paramref name="endSegment" /> is <see langword="null" />.</exception>
+        <exception cref="T:System.ArgumentOutOfRangeException">The running index of <paramref name="startSegment" /> is greater than the running index of <paramref name="endSegment" />, even though <paramref name="startSegment" /> is different to <paramref name="endSegment" />.
+
+-or-
+
+<paramref name="startSegment" /> is equal to <paramref name="endSegment" /> but <paramref name="endIndex" /> is smaller than <paramref name="startIndex" />.
+
+-or-
+
+<paramref name="startIndex" /> is greater than the length of the underlying memory block of <paramref name="startSegment" />.</exception>
```
Comment (mairaw) [inline xml/System.Buffers/ReadOnlySequence`1.xml]: unrelated to your changes: line 191 is missing a period

File: xml/System.Buffers/ReadOnlySequence`1.xml [hunk 5]
```diff
@@ -148,12 +148,22 @@ The consumer is expected to manage the lifetime of memory until <see cref="T:Sys
...
-        <param name="startIndex">To be added.</param>
-        <param name="endSegment">To be added.</param>
-        <param name="endIndex">To be added.</param>
+        <param name="startSegment">The initial node of the linked memory list.</param>
+        <param name="startIndex">The position to the start of the sequence inside <paramref name="startSegment" />.</param>
+        <param name="endSegment">The final node of the linked memory list.</param>
+        <param name="endIndex">The position to the end of the sequence inside <paramref name="endSegment" />.</param>
         <summary>Creates an instance of a <see cref="T:System.Buffers.ReadOnlySequence`1" /> from a linked memory list represented by start and end segments and the corresponding indexes in them.</summary>
         <remarks>To be added.</remarks>
+        <exception cref="T:System.ArgumentNullException"><paramref name="startSegment" /> or <paramref name="endSegment" /> is <see langword="null" />.</exception>
```
Comment (mairaw) [inline xml/System.Buffers/ReadOnlySequence`1.xml]:
```suggestion
        <exception cref="T:System.ArgumentNullException">
          <paramref name="startSegment" /> or <paramref name="endSegment" /> is <see langword="null" />.</exception>
```

File: xml/System.Buffers/ReadOnlySequence`1.xml [hunk 6]
```diff
@@ -346,10 +356,10 @@ The consumer is expected to manage the lifetime of memory until <see cref="T:Sys
         <Parameter Name="origin" Type="System.SequencePosition" />
       </Parameters>
       <Docs>
-        <param name="offset">To be added.</param>
-        <param name="origin">To be added.</param>
-        <summary>Returns a new <see cref="T:System.SequencePosition" /> at an <paramref name="offset" /> from the <paramref name="origin" />.</summary>
-        <returns>A new object that starts at the <paramref name="offset" /> position of the original object.</returns>
+        <param name="offset">The offset from the specified <paramref name="origin" /> sequence position.</param>
+        <param name="origin">A sequence position representing the point from which to initiate the offset.</param>
+        <summary>Returns a new <see cref="T:System.SequencePosition" /> at an <paramref name="offset" /> from the <paramref name="origin" /> position.</summary>
```
Comment (mairaw) [inline xml/System.Buffers/ReadOnlySequence`1.xml]:
```suggestion
        <summary>Returns a new <see cref="T:System.SequencePosition" /> starting at the specified offset from the <paramref name="origin" /> position.</summary>
```

File: xml/System.Buffers/ReadOnlySequenceSegment`1.xml (hunk 1 above)
Comment (Jozkee) [reply xml/System.Buffers/ReadOnlySequenceSegment`1.xml]: @mairaw `ReadOnlySequenceSegment` is a class.

File: xml/System.Buffers/ReadOnlySequence`1.xml (hunk 4 above)
Comment (carlossanlop) [reply xml/System.Buffers/ReadOnlySequence`1.xml]: Nice catch, thanks. Fixed.

File: xml/System.Buffers/ReadOnlySequenceSegment`1.xml [hunk 7]
```diff
@@ -47,7 +47,7 @@
       </AssemblyInfo>
       <Parameters />
       <Docs>
-        <summary>To be added.</summary>
+        <summary>Initializes a new instance of the <see cref="T:System.Buffers.ReadOnlySequenceSegment`1" /> struct.</summary>
```
Comment (Jozkee) [inline xml/System.Buffers/ReadOnlySequenceSegment`1.xml]:
```suggestion
        <summary>Initializes a new instance of the <see cref="T:System.Buffers.ReadOnlySequenceSegment`1" /> class.</summary>
```

File: xml/System.Buffers/ReadOnlySequenceSegment`1.xml (hunk 1 above)
Comment (mairaw) [reply xml/System.Buffers/ReadOnlySequenceSegment`1.xml]: Fixed. I was looking at the ReadOnlySequence source code when I provided the suggestion.

**Pull request merged**